docs/src/Popover.doc.js

```diff
@@ -1,261 +1,1316 @@
+  const isInViewport = () => {
+    const rect = viewRef && viewRef.current && viewRef.current.getBoundingClientRect();
+    const isVisible = rect
+      ? rect.top >= 0 &&
+        rect.left >= 0 &&
+        rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+        rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+      : undefined;
 
-    If you put the \`Popover\` in a portal or provider or it no longer shares
-    a relative root with the \`anchor\`, you must set \`positionRelativeToAnchor=false\` in order for it to be
-    positioned correctly relative to the body.
-  `}
-    name="anchor"
-  />,
+    if (isVisible) {
+      setOpen(true)
+    }
+  }
+
+  React.useEffect(() => {
+    isInViewport()
+    document.addEventListener('scroll',  isInViewport), [document, isInViewport]
+  });
+
+  return (
+    <React.Fragment>
+      <Box ref={viewRef} height={200}>
+        <Box ref={anchorRef} role="tablist">
+          <Tabs
+            activeTabIndex={1}
+            onChange={() => {}}
+            tabs={[{ href: "#Anchor", text: "Created"}]}
+            size="lg"
+          />
+        </Box>
+      </Box>
+      {open &&
+        <Layer>
+          <Popover
+            anchor={anchorRef.current}
+            color="blue"
+            idealDirection="down"
+            showCaret
+            onDismiss={() => {}}
+            positionRelativeToAnchor={false}
+            size={240}
+          >
+            <Box padding={3}>
+              <Flex alignItems="center" direction="column" gap={4}>
+                <Text color="white" align="center">
+                  New look! Click Created to see Pins you've published. Clicked Saved to see your saved Pins and boards.
+                </Text>
+                <Button
+                  color="white"
+                  inline
+                  onClick={() => setOpen(false)}
+                  size="lg"
+                  text="Got it!"
+                />
+              </Flex>
+            </Box>
+          </Popover>
+        </Layer>}
+    </React.Fragment>
+  )
+}`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="With Layer"
+      description={`
+Popover is tipically used within [Layer](/Layer). Layer renders Popover outside the DOM hierarchy of the parent allowing it to overlay surrounding content. Popover calculates its position based on the bounding box of the \`anchor\`. Within Layer, Popover no longer shares a relative root with the \`anchor\` and requires  \`positionRelativeToAnchor=false\` to properly calculate its position relative to the anchor element.
+
+Using \`Layer\` with Popover prevents from requiring \`z-index\` to solve stacking context conflicts. Popovers within Modals and Sheets with z-indexes don't require \`zIndex\` in \`Layer\` thanks to the built-in ScrollBoundaryContainer.
```

Comment:
Maybe "Using \`Layer\` with Popover eliminates the need to use \`z-index\` to solve stacking context conflicts."

Reply:
DONE